ingest-app/src/cmr/ingest/api/generic_documents.clj

```diff
@@ -318,64 +318,40 @@
-          publish-result (case concept-type-in-draft
-                           :collection (collections/ingest-collection provider-id native-id request)
-                           :tool (tools/ingest-tool provider-id native-id request)
-                           :service (services/ingest-service provider-id native-id request)
-                           (create-generic-document request))]
-      (if (contains? #{200 201} (:status publish-result))
-        ;;construct request to delete the draft.
-        (let [delete-request (-> request
-                                 (assoc-in [:route-params :native-id] draft-native-id)
-                                 (assoc-in [:route-params :concept-type] draft-concept-type)
-                                 (assoc-in [:params :native-id] draft-native-id)
-                                 (assoc-in [:params :concept-type] draft-concept-type))
-              delete-result (delete-generic-document delete-request)]
-          (if (= 200 (:status delete-result))
-            publish-result
-            (errors/throw-service-error
-             :bad-request
-             (format "Publishing draft is successful with info [%s]. Deleting draft failed with info [%s]."
-                     (:body publish-result) (:body delete-result)))))
-        publish-result))))
-    
-(defn- publish-variable-draft
-  "Publish a variable draft concept. i.e. Ingest the corresponding variable and
-  delete the variable draft."
-  [native-id request coll-concept-id coll-revision-id var-draft-id]
-  (let [{:keys [draft-concept-type provider-id concept-type-in-draft]}           
-        (extract-info-from-concept-id var-draft-id) ]
-    ;; Get information from metadata-db. 
-    (let [info (get-info-from-metadata-db request var-draft-id provider-id concept-type-in-draft)
-          request (:request info)
-          draft-native-id (:native-id info)
-          publish-result (variables/ingest-variable provider-id native-id request coll-concept-id coll-revision-id)]
-      (if (contains? #{200 201} (:status publish-result))
-        ;;construct request to delete the draft.
-        (let [delete-request (-> request
-                                 (assoc-in [:route-params :native-id] draft-native-id)
-                                 (assoc-in [:route-params :concept-type] draft-concept-type)
-                                 (assoc-in [:params :native-id] draft-native-id)
-                                 (assoc-in [:params :concept-type] draft-concept-type))
-              delete-result (delete-generic-document delete-request)]
-          (if (= 200 (:status delete-result))
-            publish-result
-            (errors/throw-service-error
-             :bad-request
-             (format "Publishing draft is successful with info [%s]. Deleting draft failed with info [%s]."
-                     (:body publish-result) (:body delete-result)))))
-        publish-result)))) 
+(defn- publish-draft-concept
```

Comment:
Unit test? Although... this one might require a lot of mocking

Reply:
I wrote 1 test for publish-draft, but it really just tests some if statements.  I am not going to write one for publish-draft-concept, because it won't test anything but a couple of if statements.